test/integration/singlecluster/controller/jobs/job/job_controller_test.go

```diff
@@ -3215,14 +3215,16 @@ var _ = ginkgo.Describe("Job with elastic jobs via workload-slices support", gin
 			workloads := &kueue.WorkloadList{}
 			g.Expect(k8sClient.List(ctx, workloads, client.InNamespace(testJob.Namespace))).Should(gomega.Succeed())
 			g.Expect(workloads.Items).Should(gomega.HaveLen(2))
+			newSliceIndex := -1
 			for i := range workloads.Items {
 				if workloads.Items[i].Name == testJobWorkload.Name {
 					g.Expect(workload.IsFinished(&workloads.Items[i])).Should(gomega.BeTrue())
 					continue
 				}
-				testJobWorkload = &workloads.Items[i]
+				newSliceIndex = i
 				util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, testJobWorkload)
 			}
+			testJobWorkload = &workloads.Items[newSliceIndex]
```

Comment:
```suggestion
			for i := range workloads.Items {
				if workloads.Items[i].UID == testJobWorkload.UID {
					g.Expect(workload.IsFinished(&workloads.Items[i])).Should(gomega.BeTrue())
					continue
				}
				util.ExpectWorkloadsToBeAdmitted(ctx, k8sClient, workloads.Items[i])
			}
```
 
I guess that this should be enough and safe for `"old workload is finished and new workload is admitted"`, right?